Backend/src/controllers/user.controller.ts

```diff
@@ -12,114 +12,156 @@ limitations under the License.
   const { id } = req.params;
 
-  const user = await User.findOne({ _id: id }).exec();
+  try {
+    const adminToken = await getAdminToken();
+    const response = await axios.get(
+      `http://keycloak:8080/admin/realms/jasminegraph/users/${id}`,
+      { headers: { Authorization: `Bearer ${adminToken}` } }
+    );
+
+    if (!response.data) {
+      return res.status(HTTP[404]).json({ message: `User with id "${id}" not found.` });
+    }
 
-  if (!user) {
-    return res.status(HTTP[404]).json({ message: `User with id "${id}" not found.` });
+    return res.status(HTTP[200]).json({ data: response.data });
+  } catch (err: any) {
+    console.error('[GET USER] Error:', err.response?.data || err.message);
+    return res.status(HTTP[500]).json({ message: 'Server error' });
   }
-
-  return res.status(HTTP[200]).json({ data: user });
 };
 
 const updateUser = async (req: Request, res: Response) => {
   const { id } = req.params;
   const { enabled, fullName, role } = req.body;
 
-  const user = await User.findOne({ _id: id });
-
-  if (!user) {
-    return res.status(HTTP[404]).json({ message: `User with id "${id}" not found.` });
-  }
-
   if (!fullName || !role) {
     return res.status(HTTP[422]).json({ message: 'The fields full name and role are required' });
   }
 
-  await User.updateOne({ _id: id }, { enabled, fullName, role });
+  try {
+    const adminToken = await getAdminToken();
 
-  const userUpdated = await User.findById(id);
+    const updatePayload: any = {
+      enabled,
+      firstName: fullName,
+      lastName: fullName,
+      attributes: { role: [role] }
+    };
 
-  return res.status(HTTP[200]).json({ data: userUpdated });
+    await axios.put(
+      `http://keycloak:8080/admin/realms/jasminegraph/users/${id}`,
+      updatePayload,
+      { headers: { Authorization: `Bearer ${adminToken}` } }
+    );
+
+    const updatedUserResponse = await axios.get(
+      `http://keycloak:8080/admin/realms/jasminegraph/users/${id}`,
+      { headers: { Authorization: `Bearer ${adminToken}` } }
+    );
+
+    return res.status(HTTP[200]).json({ data: updatedUserResponse.data });
+  } catch (err: any) {
+    console.error('[UPDATE USER] Error:', err.response?.data || err.message);
+    return res.status(HTTP[500]).json({ message: 'Server error' });
+  }
 };
 
-// getUserByToken will return user by accessToken in the authorization header
 const getUserByToken = async (req: Request, res: Response) => {
-  const accessToken = req.headers.authorization?.split(' ')[1];
-  const token = await Token.findOne({ accessToken });
-  if (!token) {
-    return res.status(HTTP[401]).json({ message: 'Unauthorized' });
-  }
-  const user = await User.findOne({ _id: token.userId });
-  if (!user) {
-    return res.status(HTTP[404]).json({ message: 'User not found' });
-  }
+  try {
+    const authHeader = req.headers.authorization;
+    if (!authHeader) {
+      return res.status(HTTP[401]).json({ message: 'Authorization header missing' });
```

Comment:
Even though these error messages are returned I hope the UI will show meaningful error message to the users after capturing these messages at a higher-level.